trieste/acquisition/function.py

```diff
@@ -1365,3 +1365,29 @@ def repulsion_term(
             )
 
     return acquisition
+
+
+class PredictiveVariance(SingleModelAcquisitionBuilder):
+    """
+    Builder for the determinant of the predictive covariance matrix over the batch points.
+    For a batch of size 1 it is the same as maximizing the predictive variance.
+    """
+
+    def __repr__(self) -> str:
+        """"""
+        return "PredictiveVariance()"
+
+    def prepare_acquisition_function(
+        self, dataset: Dataset, model: ProbabilisticModel
+    ) -> AcquisitionFunction:
+        """
+        :param dataset: Unused.
+        :param model: The model over the specified ``dataset``.
+        :return: The determinant of the predictive function.
+        """
+
+        def determinantcovariance(model: ProbabilisticModel, at: TensorType) -> TensorType:
+            _, covariance = model.predict_joint(at)
+            return tf.linalg.det(covariance)
+
+        return lambda at: determinantcovariance(model, at)
```

Comment:
you can use `model` from the larger scope (indeed you're already doing this in the `lambda`)
```suggestion
        def determinantcovariance(at: TensorType) -> TensorType:
            _, covariance = model.predict_joint(at)
            return tf.linalg.det(covariance)

        return determinantcovariance
```